quickstarts/microsoft.datashare/data-share-share-storage-account/main.bicep

```diff
@@ -0,0 +1,107 @@
+@description('Specify a project name that is used to generate resource names.')
+param projectName string
+
+@description('Specify the location for the resources.')
+param location string = resourceGroup().location
+
+@description('Specify an email address for receiving data share invitations.')
+param invitationEmail string
+
+@description('Specify snapshot schedule recurrence.')
+@allowed([
+  'Day'
+  'Hour'
+])
+param syncInterval string = 'Day'
+
+@description('Specify snapshot schedule start time.')
+param syncTime string = utcNow('yyyy-MM-ddTHH:mm:ssZ')
+
+var storageAccountName = '${projectName}store'
+var containerName = '${projectName}container'
+var dataShareAccountName = '${projectName}shareaccount'
+var dataShareName = '${projectName}share'
+var roleAssignmentName = guid(uniqueString(storageAccountName, storageBlobDataReaderRoleDefinitionId, dataShareAccountName))
+var inviteName = '${dataShareName}invite'
+var storageBlobDataReaderRoleDefinitionId = resourceId('Microsoft.Authorization/roleDefinitions', '2a2b9908-6ea1-4ae2-8e65-a410df84e7d1')
+
+resource sa 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'StorageV2'
+  properties: {
+    accessTier: 'Hot'
+  }
+}
+
+resource container 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-04-01' = {
+  name: '${sa.name}/default/${containerName}'
+}
+
+resource dataShareAccount 'Microsoft.DataShare/accounts@2021-08-01' = {
+  name: dataShareAccountName
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {}
+}
+
+resource dataShare 'Microsoft.DataShare/accounts/shares@2021-08-01' = {
+  parent: dataShareAccount
+  name: dataShareName
+  properties: {
+    shareKind: 'CopyBased'
+  }
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: sa
+  name: roleAssignmentName
+  properties: {
+    roleDefinitionId: storageBlobDataReaderRoleDefinitionId
+    principalId: dataShareAccount.identity.principalId
+  }
+}
+
+resource dataSet 'Microsoft.DataShare/accounts/shares/dataSets@2021-08-01' = {
+  parent: dataShare
+  name: containerName
+  kind: 'Container'
+  properties: {
+    subscriptionId: subscription().subscriptionId
+    resourceGroup: resourceGroup().name
+    storageAccountName: storageAccountName
+    containerName: containerName
+  }
+  dependsOn: [
```

Comment:
remove all the dependsOn for the roleAssignment - I don't think it's necessary since the template is working and the RA is likely not replicated.